lib/class-wp-rest-block-directory-controller.php

```diff
@@ -35,278 +35,443 @@ public function register_routes() {
+					'type'        => 'string',
+					'context'     => array( 'view' ),
+				),
+				'author_block_rating' => array(
+					'description' => __( 'The average rating of blocks published by the same author.', 'gutenberg' ),
+					'type'        => 'integer',
+					'context'     => array( 'view' ),
+				),
+				'author_block_count'  => array(
+					'description' => __( 'The number of blocks published by the same author.', 'gutenberg' ),
+					'type'        => 'integer',
+					'context'     => array( 'view' ),
+				),
+				'author'              => array(
+					'description' => __( 'The WordPress.org username of the block author.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'view' ),
+				),
+				'icon'                => array(
+					'description' => __( 'The block icon.', 'gutenberg' ),
+					'type'        => 'string',
+					'format'      => 'uri',
+					'context'     => array( 'view' ),
+				),
+				'humanized_updated'   => array(
+					'description' => __( 'The date when the block was last updated, in fuzzy human readable format.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'view' ),
+				),
+				'assets'              => array(
+					'description' => __( 'An object representing the block CSS and JavaScript assets.', 'gutenberg' ),
+					'type'        => 'array',
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+					'items'       => array(
+						'type'   => 'string',
+						'format' => 'uri',
+					),
 
-		$block->id                  = $plugin['slug'];
-		$block->rating              = $plugin['rating'] / 20;
-		$block->rating_count        = $plugin['num_ratings'];
-		$block->active_installs     = $plugin['active_installs'];
-		$block->author_block_rating = $plugin['author_block_rating'] / 20;
-		$block->author_block_count  = (int) $plugin['author_block_count'];
+				),
 
-		// Plugin's author, not author in block.json.
-		$block->author = wp_strip_all_tags( $plugin['author'] );
+			),
+		);
 
-		// Plugin's icons or icon in block.json.
-		$block->icon = isset( $plugin['icons']['1x'] ) ? $plugin['icons']['1x'] : 'block-default';
+		return $this->add_additional_fields_schema( $this->schema );
+	}
 
-		$block->assets = array();
+	/**
+	 * Retrieves the search params for the blocks collection.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @return array Collection parameters.
+	 */
+	public function get_collection_params() {
+		$query_params = parent::get_collection_params();
 
-		foreach ( $plugin['block_assets'] as $asset ) {
-			$block->assets[] = 'https://plugins.svn.wordpress.org/' . $plugin['slug'] . $asset;
-		}
+		$query_params['context']['default']  = 'view';
+		$query_params['per_page']['default'] = 3;
 
-		$block->humanized_updated = sprintf(
-			/* translators: %s: Human-readable time difference. */
-			__( '%s ago', 'gutenberg' ),
-			human_time_diff( strtotime( $plugin['last_updated'] ), time() )
+		$query_params['term'] = array(
```

Comment:
> I am able to reproduce the issue that i mentioned in my review in master now. I'm curious, how did you test?

Ack! I'm not seeing your review, did I miss it? Sorry about that, I'll work on a fix.

I tested by opening the block inserter, searching for `p3` and installing the block.

Reply:
I'm also seeing the `Unable to install this block.` message, I think because the JS expects the response to be `{"success": true}`. I don't think it's an issue with the API, it's something we should fix in the Block Directory code now that this has landed.

I'm not sure about the `filterValue` issue @StevenDufresne mentions, though I do see an initial request with an empty term, it just returns regular plugins, not block-plugins.

Reply:
Oh, and with the "unable …" message— the plugin installs correctly, it's just not registering in the directory.

Reply:
I think I didn't properly rebase which messed up my testing. PR incoming.

Reply:
I opened #23096 to fix it. Sorry about that!